### DISCUSSION TITLE
API doesn't work in electron?

### DISCUSSION THREAD

**Discussion by notwatermango** [vvo/iron-session#862] [Q&A]
Hi, I am trying to make my NextJS 14/15 app wrapped in electron. Everything is fine so far except `getIronSession` is throwing an error.

When I tried logging it back the error is just `{}`

I checked the `cookies()` from `next/header` is working fine. 

I tried setting the cookies natively and it worked. Only the `getIronSession` is not working in electron. 

My setup is a template from https://github.com/spa5k/nextjs_approuter_electron

Posting here if anyone had a clue why this doesn't work. (in dev it works, only in production it doesn't)

I was suspecting my setup in electron, but I have no clue why that doesn't work.

Why setting up cookie manually in App Router handlers work but getIronSession doesn't? Is the error related to seal unseal data failure?

**Comment by vvo** [comment]
@notwatermango Worth retrying on v9, which names what failed instead of throwing an empty object. Two things that would help here:

- `onUnsealError: (reason, error) => console.warn(reason, error)` reports a cookie that could not be read, which used to be completely silent.
- The size and serialization errors now say which field or value caused them.

Since it works in dev and not in a packaged build, the first thing I'd check is `secure: true` over plain http. Every browser refuses that cookie, and a packaged Electron app is not on https. Use `secure: process.env.NODE_ENV === "production"` and see if that is it.

v9 notes: https://github.com/vvo/iron-session/releases/tag/v9.0.0
